### DISCUSSION TITLE
How to do query group by / pivot?

### DISCUSSION THREAD

**Discussion by hjalmarlucius** [geldata/gel#3285] [Q&A] [answered]
Say I have the following schema:
```
type location {
  required property name -> str;
  multi link temperature -> temperatures;
  multi link temperatureforecasts -> temperatureforecasts;
}
type temperatures {
  required property timestep -> int32;
  required property value -> float32;
}
type temperatureforecasts {
  required property timestep -> int32;
  required property value -> array<float32>;
}
```

How can I do a query where I get `temperatures` and `temperatureforecasts` for each timestep? Effectively a join on the schema-wise unrelated attribute `timestep`.

**Comment by elprans** [comment, marked as answer]
You have to make a distinct set of timesteps and then `union` filtered subsets.  Something like this, if I got your question right:

```
for timestep in (distinct (temperatures.timestep union temperatureforecasts.timestep))
union
    {
        timestep := timestep,
        temperatures := (select temperatures {value} filter .timestep = timestep),
        forecasts := (select temperatureforecasts {value} filter .timestep = timestep),
    };
```

We are currently working on a general `GROUP` statement that will make grouping significantly more ergonomic and efficient (see [RFC 1009](https://github.com/edgedb/rfcs/blob/master/text/1009-group.rst))

**Reply by hjalmarlucius** [reply]
Outstanding! I'm evaluating which DB to use for my startup and the absence of output reshaping was a concern - any ETA for when RFC 1009 is usable with reasonable robustness (not necessarily production level)?

**Reply by hjalmarlucius** [reply]
in case others are looking: please note that the link above has been moved - it should be https://github.com/edgedb/rfcs/blob/master/text/1009-group.rst

**Reply by elprans** [reply]
The `GROUP` implementation is already being worked on and will likely appear in nightlies sometime in January, although there's no hard ETA on it yet.

**Reply by hjalmarlucius** [reply]
thanks!
